v2/pkg/mongo-driver/graceful.go

```diff
@@ -0,0 +1,24 @@
+package mongo_driver
+
+import (
+	"context"
+	"github.com/ONSdigital/log.go/log"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// Graceful represents an interface to the shutdown method
+type Graceful interface {
+	shutdown(ctx context.Context, mongoClient *mongo.Client, closedChannel chan bool)
+}
+
+type graceful struct{}
+
+func (t graceful) shutdown(ctx context.Context, mongoClient *mongo.Client, closedChannel chan bool) {
+	err := mongoClient.Disconnect(ctx)
```

Comment:
disconnect with timeout ?

Reply:
@s-xton  this deadline is there in the context that is passed to this method.